dataplicity/m2m/packetbase.py

```diff
@@ -31,6 +31,9 @@ def __new__(mcs, name, bases, attrs):
             if packet_cls.type >= 0:
                 assert packet_cls.type not in packet_cls.registry, "packet type {!r} has already been registered".format(packet_cls, type)
                 packet_cls.registry[packet_cls.type] = packet_cls
+                # register the packet type in base class so we can have a
+                # generic interface of accessing it.
+                bases[0].registry[packet_cls.type] = packet_cls
```

Comment:
You can do the following:

```
M2MPacket.from_bytes(b'...')
```

Which will return an instance of the appropriate class. M2MPacket is defined in `packets.py`.